core/deployment/src/main/java/io/quarkus/deployment/builditem/JniBuildItem.java

```diff
@@ -0,0 +1,31 @@
+package io.quarkus.deployment.builditem;
+
+import java.util.Optional;
+
+import io.quarkus.builder.item.SimpleBuildItem;
+
+public final class JniBuildItem extends SimpleBuildItem {
+
+    private Optional<Boolean> enableJni;
```

Comment:
Build items need to be immutable. The producer/consumer model does not really allow for mutation, so this might work if the steps that modify it also produce other items and the order happens to work out, however it is a bug waiting to happen.

Instead this should be a MultiBuildItem and extensions should just produce an instance if they require JNI. If this list is empty then nothing requires JNI, otherwise it should be enabled.